Riot/Modules/Common/Recents/RecentsViewController.m

```diff
@@ -1602,10 +1602,11 @@ - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath
         
         if (roomIdOrAlias.length)
         {
-            // Open the room or preview it
-            NSString *fragment = [NSString stringWithFormat:@"/room/%@", [MXTools encodeURIComponent:roomIdOrAlias]];
-            NSURL *url = [NSURL URLWithString:[MXTools permalinkToRoom:fragment]];
-            [[AppDelegate theDelegate] handleUniversalLinkFragment:fragment fromURL:url];
+            // Create a permalink to open or preview the room.
+            NSString *permalink = [MXTools permalinkToRoom:roomIdOrAlias];
+            NSURL *permalinkURL = [NSURL URLWithString:permalink];
+            UniversalLink *link = [[UniversalLink alloc] initWithUrl:permalinkURL];
+            [[AppDelegate theDelegate] handleUniversalLinkFragment:permalinkURL.fragment fromLink:link];
```

Comment:
Same as the other, it would be easier to call like:

```suggestion
            [[AppDelegate theDelegate] handleUniversalLinkURL:permalinkURL];
```